PackageSourceMappingExample/README.md

```diff
@@ -1,9 +1,322 @@
+
+Older tooling will ignore the Package Source Mapping configuration. To use this feature, ensure all your build environments use compatible tooling versions.
+
+Package Source Mappings will apply to all project types - including .NET Framework - as long as compatible tooling is used for build and restore.
 
 Here's an [example](https://github.com/NuGet/NuGet.Client/blob/dev-kartheekp-ms-dogfood-namespaces/NuGet.Config) of how the NuGet.Config for the NuGet.Client repo looks like in our dogfooding efforts.
+
+## Package Source Mapping rules
+
+1. Two types of package ID patterns are supported:
+
+    a. `NuGet.*` - Package prefixes. Must end with a `*`, which may match 0 or more characters. `*` is the broadest valid prefix that matches all package IDs, but will have the lowest precedence by default. `NuGet*` is also valid and will match package IDs `NuGet`, `NuGetFoo`, and `NuGet.Bar`.
+    
+    b. `NuGet.Common` - Exact package IDs.
+
+2. Any requested package ID must map to one or more sources by matching a defined package ID pattern. In other words, once you have defined a `packageSourceMapping` element you must explicitly define which sources *every* package - *including transitive packages* - will be restored from.
+
+    a. Both top level (directly installed) *and transitive* packages must match defined patterns. There is no requirement that a top level package and its dependencies come from the same source.
+    
+    b. The same ID pattern can be defined on multiple sources, allowing matching package IDs to be restored from any of the feeds that define the pattern. However, this isn't recommended due to the impact on restore predictability (a given package could come from multiple sources).
+
+3. When multiple unique patterns match a package ID, the most specific (longest) match will be preferred. 
+
+    a. Exact package ID patterns always have the highest precedence while the generic `*` always has the lowest precedence. For an example package ID `NuGet.Common`, the following package ID patterns are ordered from highest to lowest precedence: `NuGet.Common`, `NuGet.*`, `*`. 
+
+4. Package Source Mapping settings are applied following [nuget.config precedence rules](https://docs.microsoft.com/nuget/consume-packages/configuring-nuget-behavior#how-settings-are-applied) when multiple `nuget.config` files at various levels (machine-level, user-level, repo-level) are present. 
+
+> Important: When the requested package already exists in the global packages folder, no source look-up will happen and the mappings will be ignored. Declare a [global packages folder for your repo](https://docs.microsoft.com/nuget/reference/nuget-config-file#config-section) to gain the full security benefits of this feature. Work to improve the experience with the default global packages folder in planned for a next iteration.
+
+## Package Source Mapping onboarding
+
+To fully onboard your repository take the following steps:

```

Comment:
Maybe change the wording, to may take the following. 

I think the language should be, here's a way you can do it, rather than here's how you do it. 

Wyt? I'm certainly no authority here :)

Reply:
Removed